### ISSUE TITLE
Custom degrees of freedom argument for the `PowerDivergenceTest`

### ISSUE THREAD

**Issue by BERENZ** [JuliaStats/HypothesisTests.jl#266]
Is it possible to add a custom parameter for the degrees of freedom in the `PowerDivergenceTest` function? This may be useful for the goodness of fit tests ($G^2, \chi^2$) as the current implementation does not account for the number of parameres of the reference distribution. I think that implementation proposed in [scipy.stats.power_divergence](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.power_divergence.html) seems be the right way to do so (`ddof` parameter). Now, the user needs to remember to correct p-value after the test.

**Comment by nalimilan** [comment]
Makes sense. Could you provide other references about this adjustment? Would you be willing to make a pull request?

**Comment by BERENZ** [comment]
Sure, I will make a pull request to the end of this week.

**Comment by kunzaatko** [comment]
Hello @BERENZ. May I ask about the state of this issue? It looks like you already have a commit in your fork for this. It there something else missing that needs to be done before the upstreaming? I can help with something if necessary.
